azurerm/internal/services/securitycenter/server_vulnerability_assessment_resource.go

```diff
@@ -0,0 +1,176 @@
+		return tf.ImportAsExistsError("azurerm_server_vulnerability_assessment", *vulnerabilityAssessment.ID)
+	}
+
+	vulnerabilityAssessment, err = client.CreateOrUpdate(ctx, resourceGroup, provider, resourceType, name)
+	if err != nil {
+		return fmt.Errorf("create Server Vulnerability Assessment for %q/%q: %+v", resourceGroup, name, err)
+	}
+
+	if vulnerabilityAssessment.ID != nil {
+		d.SetId(*vulnerabilityAssessment.ID)
+	}
+
+	// Take some time for resource to be visible after creating. Hence the retry logic
+	readError := resourceServerVulnerabilityAssessmentRead(d, meta)
+	for retry := 1; d.Id() == "" && retry < 11; retry++ {
+		time.Sleep(time.Duration(retry) * time.Second)
+		readError = resourceServerVulnerabilityAssessmentRead(d, meta)
+	}
+	return readError
+}
+
+func resourceServerVulnerabilityAssessmentRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	vmId := d.Get("virtual_machine_id").(string)
+	var idToParse string
+	if vmId != "" {
+		idToParse = vmId
+	} else {
+		idToParse = d.Id()
+	}
+	resourceGroup, provider, resourceType, name, id, err := parseVirtualMacineId(idToParse)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, provider, resourceType, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("Server Vulnerability Assessment was not found for %q/%q: %+v", resourceGroup, name, err)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Server Vulnerability Assessment status for %q/%q: %+v", resourceGroup, name, err)
+	}
+
+	d.SetId(*resp.ID)
+	if vmId == "" {
+		// Need to set virtual_machine_id when we do a import
+		_ = d.Set("virtual_machine_id", "/subscriptions/"+id.SubscriptionID+
+			"/resourceGroups/"+id.ResourceGroup+
+			"/providers/"+provider+"/"+resourceType+"/"+id.Path[resourceType])
+	}
+
+	return nil
+}
+
+func resourceServerVulnerabilityAssessmentDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup, provider, resourceType, name, _, err := parseVirtualMacineId(d.Get("virtual_machine_id").(string))
+	if err != nil {
+		return err
+	}
+
+	// Can not delete if still in provisioning state. Wait for it to complete.
+	for retry := 1; retry <= 20; retry++ {
+		response, err := client.Get(ctx, resourceGroup, provider, resourceType, name)
+		if err != nil || response.ProvisioningState != security.ProvisioningState1Provisioning {
+			break
+		}
+		time.Sleep(time.Duration(retry) * time.Second)
+	}
+
+	response, err := client.Delete(ctx, resourceGroup, provider, resourceType, name)
+	if err != nil {
+		if utils.ResponseWasStatusCode(response, 202) {
```

Comment:
I would file this as a bug against the upstream SDK. If the API gives a 202 on "success" then it should probably be added to the list of accepted codes in DeleteResponder.

Reply:
[Bug added to the SDK](https://github.com/Azure/azure-sdk-for-go/issues/14120)